src/main/scala/tile/RocketTile.scala

```diff
@@ -137,6 +137,14 @@ class RocketTile(
     }
   }
 
+  class RocketLogicalTree extends LogicalTree {
```

Comment:
Instead of calling `cpuDevice.getOMComponents` below, could you instead move the code from `getOMRocketCores` into `getOMComponents` here, make the other OM functions be methods on `RocketLogicalTree`, and finally make `RocketLogicalTree` take `(rocketParams: RocketTileParams, xlen: Int, ...)` as constructor arguments? 


That way, rather than adding another layer of shallow indirection, you would
1) be making the API for the OM metadata layer more explicit
2) encapsulate all functions it uses to generate information in one class
3) decouple yourself from the `Device`/`Description`/`Property` API, which AFAICT you aren't actually using / don't actually need

Reply:
I changed the API